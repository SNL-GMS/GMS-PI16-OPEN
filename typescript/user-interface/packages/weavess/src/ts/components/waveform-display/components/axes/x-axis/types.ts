export interface XAxisProps {
  /** Epoch seconds end */
  endTimeSecs: number;

  /** Epoch seconds start */
  startTimeSecs: number;

  /** Add border to top */
  borderTop: boolean;

  /** Label width in px */
  labelWidthPx: number;

  /** Scrollbar width in px */
  scrollbarWidthPx: number;

  /** (optional) x-axis label  */
  label?: string;

  /** Call back to get the current view range */
  getViewRange(): [number, number];
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface XAxisState {}
