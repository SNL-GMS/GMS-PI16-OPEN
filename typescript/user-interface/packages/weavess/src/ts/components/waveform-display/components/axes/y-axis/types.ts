export interface YAxisProps {
  /** The height in percentage */
  heightInPercentage: number;

  /** Max amplitude as a number */
  maxAmplitude: number;

  /** Min amplitude as a number */
  minAmplitude: number;

  /** the y axis ticks */
  yAxisTicks?: number[];
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface YAxisState {}
