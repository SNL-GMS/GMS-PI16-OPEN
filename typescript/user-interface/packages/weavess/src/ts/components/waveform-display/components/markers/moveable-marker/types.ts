import { WeavessTypes } from '@gms/weavess-core';

export interface MoveableMarkerProps {
  /** The marker config */
  marker: WeavessTypes.Marker;

  /** The associated start marker (usually only for a selection window) */
  associatedStartMarker?: WeavessTypes.Marker;

  /** The associated end marker (usually only for a selection window) */
  associatedEndMarker?: WeavessTypes.Marker;

  /** Percentage Location 0-100 as number */
  percentageLocation: number;

  /** Label Width in px */
  labelWidthPx: number;

  /** react key but called name since key was causing fortify issues */
  name?: string;

  /** Start and end of the entire time range */
  timeRange(): WeavessTypes.TimeRange;

  /** current view port of the time range */
  viewTimeRange(): WeavessTypes.TimeRange;

  /** Returns container client width */
  containerClientWidth(): number;

  /** Returns the viewPort client width */
  viewportClientWidth(): number;

  /**
   * (optional) updates the location of the marker
   *
   * @param marker the marker
   */
  onUpdateMarker?(marker: WeavessTypes.Marker): void;

  /**  */
  updateTimeWindowSelection?(): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface MoveableMarkerState {}
