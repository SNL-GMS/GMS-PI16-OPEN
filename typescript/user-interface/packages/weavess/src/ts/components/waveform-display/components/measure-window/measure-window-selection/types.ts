import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';
import * as React from 'react';

export interface MeasureWindowSelectionListenerProps {
  displayTimeRange: WeavessTypes.TimeRange;
  hotKeys: WeavessTypes.HotKeysConfiguration;
  isMeasureWindowEnabled(): boolean;
  children(measureWindowParams: {
    contentRenderer: JSX.Element;
    onMouseDown: (e: React.MouseEvent<HTMLDivElement>) => void;
  }): JSX.Element | null;
  toast(
    message: string,
    intent?: Intent | undefined,
    icon?: IconName | undefined,
    timeout?: number | undefined
  ): void;
  updateMeasureWindowPanel(
    timeRange: WeavessTypes.TimeRange,
    removeMeasureWindowSelection: () => void
  ): void;
  computeTimeSecsFromMouseXPixels(mouseXPx: number): number;
}
