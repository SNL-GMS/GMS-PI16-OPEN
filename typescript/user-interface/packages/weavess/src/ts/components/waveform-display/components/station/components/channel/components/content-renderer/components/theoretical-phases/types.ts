import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';

export interface TheoreticalPhasesProps {
  /** Station Id as string */
  stationId: string;

  /** The theoretical phase windows */
  theoreticalPhaseWindows: WeavessTypes.TheoreticalPhaseWindow[] | undefined;

  /** Boolean is default channel */
  isDefaultChannel: boolean;

  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Epoch seconds end time */
  displayEndTimeSecs: number;

  /** (optional) callback events Ex on waveform click */
  events?: WeavessTypes.ChannelContentEvents;

  /**
   * Call back for toast, which is the notification pop up
   *
   * @param message message to be displayed as a string
   * @param intent (optional) NONE, PRIMARY, WARNING, DANGER ex NONE = "none"
   * @param icon (optional) the way it looks
   * @param timeout (optional) time before message disappears
   */
  toast(message: string, intent?: Intent, icon?: IconName, timeout?: number): void;

  /**
   * Returns the time in seconds for the given clientX.
   *
   * @param clientX The clientX
   *
   * @returns The time in seconds; undefined if clientX is
   * out of the channel's bounds on screen.
   */
  getTimeSecsForClientX(clientX: number): number | undefined;

  /**
   * Toggle display of the drag indicator for this channel
   *
   * @param show True to show drag indicator
   * @param color The color of the drag indicator
   */
  toggleDragIndicator(show: boolean, color: string): void;

  /**
   * Set the position for the drag indicator
   *
   * @param clientX The clientX
   */
  positionDragIndicator(clientX: number): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface TheoreticalPhasesState {}
