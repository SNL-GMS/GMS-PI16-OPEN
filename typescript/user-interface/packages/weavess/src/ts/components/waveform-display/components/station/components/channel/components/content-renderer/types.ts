import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';
import React from 'react';

import { PositionConverters } from '../../../../../../../../util/types';

export interface ContentRendererProps {
  /** Station Id as string */
  stationId: string;

  /** The description */
  description?: string;

  /** The description label color */
  descriptionLabelColor?: string;

  /** Channel Id as string */
  channelId: string;

  /** Boolean is default channel */
  isDefaultChannel: boolean;

  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Boolean is default channel */
  displayEndTimeSecs: number;

  /** Web Workers */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  workerRpcs: any[];

  /** The signal detections */
  signalDetections: WeavessTypes.PickMarker[] | undefined;

  /** The predicted phases */
  predictedPhases: WeavessTypes.PickMarker[] | undefined;

  /** Should the user be able to modify the predicted phase */
  disablePredictedPhaseModification: boolean;

  /** Should the user be able to modify the predicted phase */
  disableSignalDetectionModification: boolean;

  /** The theoretical phase windows */
  theoreticalPhaseWindows: WeavessTypes.TheoreticalPhaseWindow[] | undefined;

  /** Collection of markers */
  markers?: WeavessTypes.Markers;

  /** The selections */
  selections: WeavessTypes.Selections;

  /** (optional) callback events */
  events?: WeavessTypes.ChannelContentEvents;

  /** Configuration for weavess */
  initialConfiguration: WeavessTypes.Configuration;

  /** React elements to render in the content-renderer-content area */
  contentRenderers: React.ReactElement[];

  // Functions

  converters: PositionConverters;

  /** Ref to the html canvas element */
  canvasRef(): HTMLCanvasElement | null;

  /**
   * Call back for toast, which is the notification pop up
   *
   * @param message message to be displayed as a string
   * @param intent (optional) NONE, PRIMARY, WARNING, DANGER ex NONE = "none"
   * @param icon (optional) the way it looks
   * @param timeout (optional) time before message disappears
   */
  toast(message: string, intent?: Intent, icon?: IconName, timeout?: number): void;

  /**
   * Sets the Y axis bounds
   *
   * @param min minimum bound as a number
   * @param max Maximum bound as a number
   */
  setYAxisBounds(min: number, max: number);

  /**
   * @returns current view range as [0,1]
   */
  getViewRange(): [number, number];

  /** Issues a rerender of the graphics */
  renderWaveforms(): void;

  /**
   * get the currently displayed viewTimeInterval
   * (the startTime and endTime of the currently displayed view of the waveforms)
   */
  getCurrentViewRangeInSeconds(): WeavessTypes.TimeRange;

  /**
   * Mouse move event
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onMouseMove(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * Mouse down event
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onMouseDown(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * onMouseUp event handler
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onMouseUp(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * onContextMenu event handler
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onContextMenu(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * onKeyDown event handler
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   * @param timeForMouseXPosition the time for the current mouse position X
   */
  onKeyDown(e: React.KeyboardEvent<HTMLDivElement>): void;

  /**
   * (optional) Updates the measure window
   *
   * @param stationId station id as a string
   * @param channel channel config as a Channel
   * @param startTimeSecs epoch start time secs
   * @param endTimeSecs epoch end time secs
   * @param isDefaultChannel boolean
   * @param removeSelection removed measure window selection div
   */
  updateMeasureWindow?(
    stationId: string,
    channel: WeavessTypes.Channel,
    startTimeSecs: number,
    endTimeSecs: number,
    isDefaultChannel: boolean,
    removeSelection: () => void
  ): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ContentRendererState {}
