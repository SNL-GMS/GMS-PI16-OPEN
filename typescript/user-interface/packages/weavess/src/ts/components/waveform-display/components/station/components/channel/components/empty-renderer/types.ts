export interface EmptyRendererProps {
  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Boolean is default channel */
  displayEndTimeSecs: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyRendererState {}
