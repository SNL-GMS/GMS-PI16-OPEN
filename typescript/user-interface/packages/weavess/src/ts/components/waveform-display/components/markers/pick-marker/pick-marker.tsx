/* eslint-disable react/destructuring-assignment */
import { Icon, Intent } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { WeavessMessages } from '@gms/weavess-core';
import React from 'react';

import { calculateLeftPercent } from '../../../utils';
import { PickMarkerProps, PickMarkerState } from './types';

const ICON_SIZE_PX = 14;
/**
 * An interactive marker, that is configurable, and can have specific events.
 */
export class PickMarker extends React.PureComponent<PickMarkerProps, PickMarkerState> {
  /** container reference */
  private containerRef: HTMLDivElement | null;

  /** line reference */
  private lineRef: HTMLDivElement | null;

  /** label reference */
  private labelRef: HTMLDivElement | null;

  /**
   * Constructor
   *
   * @param props props as PickMarkerProps
   */
  public constructor(props: PickMarkerProps) {
    super(props);
    this.state = {
      position: this.props.position
    };
  }

  // ******************************************
  // BEGIN REACT COMPONENT LIFECYCLE METHODS
  // ******************************************

  /**
   * Called immediately after updating occurs. Not called for the initial render.
   *
   * @param prevProps the previous props
   * @param prevState the previous state
   */
  public componentDidUpdate(prevProps: PickMarkerProps): void {
    if (!this.lineRef) return;

    if (prevProps.position !== this.props.position) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({ position: this.props.position });
    } else {
      // if the color changes, flash animation
      // eslint-disable-next-line no-lonely-if
      if (prevProps.color !== this.props.color) {
        this.lineRef.style.borderColor = this.props.color;
        setTimeout(() => {
          if (!this.lineRef) return;

          this.lineRef.style.borderColor = this.props.color;
          this.lineRef.style.transition = 'border-color 0.5s ease-in';
          setTimeout(() => {
            if (!this.lineRef) return;
            this.lineRef.style.transition = '';
            // eslint-disable-next-line @typescript-eslint/no-magic-numbers, @typescript-eslint/indent
          }, 500);
          // eslint-disable-next-line @typescript-eslint/no-magic-numbers, @typescript-eslint/indent
        }, 500);
      }
    }
  }

  /**
   * Catches exceptions generated in descendant components.
   * Unhandled exceptions will cause the entire component tree to un-mount.
   *
   * @param error the error that was caught
   * @param info the information about the error
   */
  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
  public componentDidCatch(error, info): void {
    // eslint-disable-next-line no-console
    console.error(`Weavess Pick Marker Error: ${error} : ${info}`);
  }

  // ******************************************
  // END REACT COMPONENT LIFECYCLE METHODS
  // ******************************************

  // eslint-disable-next-line react/sort-comp, complexity
  public render(): JSX.Element {
    // show the pointer cursor if the click or drag events are defined
    const cursor: React.CSSProperties =
      this.props.onClick || this.props.onDragEnd ? { cursor: 'pointer' } : { cursor: 'auto' };

    return (
      // eslint-disable-next-line jsx-a11y/no-static-element-interactions
      <div
        className="pick-marker"
        ref={ref => {
          this.containerRef = ref;
        }}
        onMouseDown={this.props.onDragEnd ? this.onMouseDown : undefined}
        onMouseUp={this.props.onDragEnd ? this.onMouseUp : undefined}
      >
        <div
          className="pick-marker-pick"
          data-cy-predicted={`pick-marker-pick-predicted-${this.props.predicted}`}
          ref={ref => {
            this.lineRef = ref;
          }}
          style={{
            borderLeft: `1.5px solid ${this.props.color}`,
            bottom: this.props.predicted ? '10%' : '55%',
            // eslint-disable-next-line @typescript-eslint/no-magic-numbers
            left: `${this.state.position}%`,
            top: this.props.predicted ? '55%' : '10%',
            boxShadow: this.props.isSelected ? `0px 0px 10px 3px ${this.props.color}` : 'initial',
            filter: this.props.filter
          }}
        />
        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
        <div
          className="pick-marker-label"
          data-cy={`pick-marker-${this.props.id}`}
          data-cy-color={`pick-marker-${this.props.color}`}
          data-cy-is-predicted-phase={this.props.predicted}
          data-cy-phase={this.props.label}
          data-cy-style-left={this.state.position}
          onClick={this.props.onClick ? this.onClick : undefined}
          onContextMenu={this.props.onContextMenu ? this.onContextMenu : undefined}
          style={{
            bottom: this.props.predicted ? '10%' : 'initial',
            left: `calc(4px + ${this.state.position}%)`,
            top: this.props.predicted ? 'initial' : '10%',
            color: this.props.color,
            filter: this.props.filter,
            ...cursor
          }}
          ref={ref => {
            this.labelRef = ref;
          }}
        >
          {this.props.label}
          {this.props.isConflicted ? (
            <Icon
              icon={IconNames.ISSUE}
              iconSize={ICON_SIZE_PX}
              intent={Intent.DANGER}
              className="pick-marker__conflict"
            />
          ) : null}
        </div>
      </div>
    );
  }

  /**
   * onClick event handler for signal detections
   *
   * @param e
   */
  private readonly onClick = (e: React.MouseEvent<HTMLDivElement>): void => {
    // prevent propagation of these events so that the underlying channel click doesn't register
    e.stopPropagation();
    if (this.props.onClick) {
      this.props.onClick(e, this.props.id);
    }
  };

  /**
   * onContextMenu menu event handler for signal detections
   *
   * @param e
   */
  private readonly onContextMenu = (e: React.MouseEvent<HTMLDivElement>): void => {
    e.stopPropagation();
    if (this.props.onContextMenu) {
      this.props.onContextMenu(e, this.props.channelId, this.props.id);
    }
  };

  /**
   * onMouseDown event handler for signal detections
   *
   * @param e
   */
  private readonly onMouseDown = (e: React.MouseEvent<HTMLDivElement>): void => {
    // prevent propagation of these events so that the underlying channel click doesn't register
    e.stopPropagation();
    // if context-menu, don't trigger
    if (e.button === 2) return;

    if (this.props.disableModification) {
      if (this.props.isConflicted) {
        this.props.toast(WeavessMessages.signalDetectionInConflict);
      } else if (!this.props.predicted) {
        this.props.toast(WeavessMessages.signalDetectionModificationDisabled);
      } else {
        this.props.toast(WeavessMessages.predictedPhaseModificationDisabled);
      }
    } else {
      const start = e.clientX;
      let currentPos = e.clientX;
      let isDragging = false;
      let diff = 0;

      const onMouseMove = (event: MouseEvent) => {
        if (!this.containerRef) return;

        currentPos = event.clientX;
        diff = Math.abs(currentPos - start);
        // begin drag if moving more than 1 pixel
        if (diff > 1 && !isDragging) {
          isDragging = true;
          if (this.labelRef) {
            this.labelRef.style.filter = 'brightness(0.5)';
          }
          if (this.lineRef) {
            this.lineRef.style.filter = 'brightness(0.5)';
          }
          this.props.toggleDragIndicator(true, this.props.color);
        }
        if (isDragging) {
          this.props.positionDragIndicator(currentPos);
        }
      };

      const onMouseUp = (event: MouseEvent) => {
        event.stopPropagation();
        if (!this.containerRef) return;

        if (isDragging) {
          this.props.toggleDragIndicator(false, this.props.color);
          if (this.labelRef) {
            this.labelRef.style.filter = this.props.filter ? this.props.filter : 'initial';
          }
          if (this.lineRef) {
            this.lineRef.style.filter = this.props.filter ? this.props.filter : 'initial';
          }
          const time = this.props.getTimeSecsForClientX(currentPos);
          if (time) {
            this.setState(
              {
                position: calculateLeftPercent(
                  time,
                  this.props.startTimeSecs,
                  this.props.endTimeSecs
                )
              },
              () => {
                if (this.props.onDragEnd) {
                  this.props.onDragEnd(this.props.id, time);
                }
              }
            );
          }
        }
        document.body.removeEventListener('mousemove', onMouseMove);
        document.body.removeEventListener('mouseup', onMouseUp);
      };

      document.body.addEventListener('mousemove', onMouseMove);
      document.body.addEventListener('mouseup', onMouseUp);
    }
  };

  /**
   * onMouseUp event handler for signal detections
   *
   * @param e
   */
  private readonly onMouseUp = (e: React.MouseEvent<HTMLDivElement>): void => {
    e.stopPropagation();
  };
}
