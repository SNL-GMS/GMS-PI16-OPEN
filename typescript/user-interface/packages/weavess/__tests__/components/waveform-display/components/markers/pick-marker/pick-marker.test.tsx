/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import * as Enzyme from 'enzyme';
import * as React from 'react';

import { PickMarker } from '../../../../../../src/ts/components/waveform-display/components/markers/pick-marker/pick-marker';
import { PickMarkerProps } from '../../../../../../src/ts/components/waveform-display/components/markers/pick-marker/types';

const props: PickMarkerProps = {
  id: 'my-id',
  channelId: 'my-channel',
  startTimeSecs: 500,
  endTimeSecs: 600,
  position: 8,
  label: 'my pick marker',
  color: 'ff000',
  filter: 'none',
  timeSecs: 540,
  predicted: false,
  isConflicted: false,
  isSelected: false,
  disableModification: false,
  onClick: jest.fn(),
  toast: jest.fn(),
  getTimeSecsForClientX: jest.fn(() => 2),
  toggleDragIndicator: jest.fn(),
  positionDragIndicator: jest.fn(),
  onContextMenu: jest.fn(),
  onDragEnd: jest.fn()
};

const event: any = {
  stopPropagation: jest.fn(),
  target: {
    offsetLeft: 5
  },
  nativeEvent: {
    offsetX: 200,
    offsetY: 180
  }
};

const wrapper = Enzyme.mount(<PickMarker {...props} />);
const instance: PickMarker = wrapper.find(PickMarker).instance() as PickMarker;

describe('Weavess PickMarker Marker', () => {
  it('Weavess Pick Marker to be defined', () => {
    expect(PickMarker).toBeDefined();
  });

  it('shallow renders', () => {
    const shallow = Enzyme.shallow(<PickMarker {...props} />);
    expect(shallow).toMatchSnapshot();
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('componentDidUpdate', () => {
    const spy = jest.spyOn(instance, 'componentDidUpdate');

    instance.componentDidUpdate({
      ...props
    });
    expect(spy).toBeCalledTimes(1);

    instance.componentDidUpdate({
      ...props,
      position: 22
    });
    expect(spy).toBeCalledTimes(2);

    instance.componentDidUpdate({
      ...props,
      color: '0000ff'
    });
    expect(spy).toBeCalledTimes(3);
  });

  it('componentDidCatch', () => {
    const spy = jest.spyOn(instance, 'componentDidCatch');
    instance.componentDidCatch(new Error('error'), { componentStack: undefined });
    expect(spy).toBeCalled();
  });

  it('onClick', () => {
    const pm: any = new PickMarker(props);
    const spy = jest.spyOn(pm, 'onClick');
    pm.onClick(event);
    expect(spy).toHaveBeenCalled();
  });

  it('onContextMenu', () => {
    const pm: any = new PickMarker(props);
    const spy = jest.spyOn(pm, 'onContextMenu');
    pm.onContextMenu(event);
    expect(spy).toHaveBeenCalled();
  });

  it('onMouseDown', () => {
    let pm: any = new PickMarker(props);
    let spy = jest.spyOn(pm, 'onMouseDown');
    pm.onMouseDown(event);
    expect(spy).toHaveBeenCalled();

    pm = new PickMarker({ ...props, disableModification: true, isConflicted: true });
    spy = jest.spyOn(pm, 'onMouseDown');
    pm.onMouseDown(event);
    expect(spy).toHaveBeenCalled();

    pm = new PickMarker({ ...props, disableModification: true, predicted: true });
    spy = jest.spyOn(pm, 'onMouseDown');
    pm.onMouseDown(event);
    expect(spy).toHaveBeenCalled();

    pm = new PickMarker({ ...props, disableModification: true });
    spy = jest.spyOn(pm, 'onMouseDown');
    pm.onMouseDown(event);
    expect(spy).toHaveBeenCalled();
  });

  it('onMouseUp', () => {
    const pm: any = new PickMarker(props);
    const spy = jest.spyOn(pm, 'onMouseUp');
    pm.onMouseUp(event);
    expect(spy).toHaveBeenCalled();
  });
});
