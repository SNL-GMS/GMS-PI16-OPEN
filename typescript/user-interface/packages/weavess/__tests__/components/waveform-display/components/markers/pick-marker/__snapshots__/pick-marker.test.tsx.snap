// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess PickMarker Marker renders 1`] = `
<PickMarker
  channelId="my-channel"
  color="ff000"
  disableModification={false}
  endTimeSecs={600}
  filter="none"
  getTimeSecsForClientX={[MockFunction]}
  id="my-id"
  isConflicted={false}
  isSelected={false}
  label="my pick marker"
  onClick={[MockFunction]}
  onContextMenu={[MockFunction]}
  onDragEnd={[MockFunction]}
  position={8}
  positionDragIndicator={[MockFunction]}
  predicted={false}
  startTimeSecs={500}
  timeSecs={540}
  toast={[MockFunction]}
  toggleDragIndicator={[MockFunction]}
>
  <div
    className="pick-marker"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="pick-marker-pick"
      data-cy-predicted="pick-marker-pick-predicted-false"
      style={
        Object {
          "borderLeft": "1.5px solid ff000",
          "bottom": "55%",
          "boxShadow": "initial",
          "filter": "none",
          "left": "8%",
          "top": "10%",
        }
      }
    />
    <div
      className="pick-marker-label"
      data-cy="pick-marker-my-id"
      data-cy-color="pick-marker-ff000"
      data-cy-is-predicted-phase={false}
      data-cy-phase="my pick marker"
      data-cy-style-left={8}
      onClick={[Function]}
      onContextMenu={[Function]}
      style={
        Object {
          "bottom": "initial",
          "color": "ff000",
          "cursor": "pointer",
          "filter": "none",
          "left": "calc(4px + 8%)",
          "top": "10%",
        }
      }
    >
      my pick marker
    </div>
  </div>
</PickMarker>
`;

exports[`Weavess PickMarker Marker shallow renders 1`] = `
<div
  className="pick-marker"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
>
  <div
    className="pick-marker-pick"
    data-cy-predicted="pick-marker-pick-predicted-false"
    style={
      Object {
        "borderLeft": "1.5px solid ff000",
        "bottom": "55%",
        "boxShadow": "initial",
        "filter": "none",
        "left": "8%",
        "top": "10%",
      }
    }
  />
  <div
    className="pick-marker-label"
    data-cy="pick-marker-my-id"
    data-cy-color="pick-marker-ff000"
    data-cy-is-predicted-phase={false}
    data-cy-phase="my pick marker"
    data-cy-style-left={8}
    onClick={[Function]}
    onContextMenu={[Function]}
    style={
      Object {
        "bottom": "initial",
        "color": "ff000",
        "cursor": "pointer",
        "filter": "none",
        "left": "calc(4px + 8%)",
        "top": "10%",
      }
    }
  >
    my pick marker
  </div>
</div>
`;
